### PR TITLE
WIP: Prometheus metrics support.

### PR DESCRIPTION
**Pull request by djpiper28**

### CODE REVIEW

**Comment by IzStriker** [comment]
Would be nice if you could get the size of the cache and the number of requests in the last 24 hours, but it might take some work. Happy to help tho

**Comment by djpiper28** [comment]
> Would be nice if you could get the size of the cache and the number of requests in the last 24 hours, but it might take some work. Happy to help tho

That's a good idea, I will look into it.

File: WakaBot.Web/Program.cs [hunk 1]
```diff
@@ -22,6 +25,22 @@
     return "Up time is " + stopwatch.Elapsed;
 });
 
+app.MapGet("/metrics", () =>
+{
+    StringBuilder sb = new StringBuilder();
+    sb.AppendLine("waka_uptime " + stopwatch.Elapsed);
+    sb.AppendLine("waka_memory_usage " + Process.GetCurrentProcess().PrivateMemorySize64);
+
+    var database = app.Services.GetService<WakaContext>();
```
Comment (IzStriker) [inline WakaBot.Web/Program.cs:34]: @djpiper28 WakaBot.Web and WakaBot.Core has two different instances of DI. To get this to work you may first need to access WakaBotService via DI, then access the bots services from there.

**Comment by IzStriker** [comment]
Gonna rewrite wakabot DI to make this work

**Pull request merged**